sdk/src/main/scala/com/horizen/api/http/ApiResponseUtil.scala

```diff
@@ -26,17 +29,14 @@ object ApiResponseUtil {
   def toResponseAsJava(response: ApiResponse): server.Route = {
     response match {
       case _: SuccessResponse => Directives.complete(StatusCodes.OK, HttpEntities.create(ContentTypes.APPLICATION_JSON, SerializationUtil.serializeWithResult(response)))
-      case e: ErrorResponse =>
-        e.exception match {
-          case Some(thr) =>
-            val msg = thr.getMessage
-            if (msg != null && !msg.isEmpty)
-              Directives.complete(StatusCodes.OK, HttpEntities.create(ContentTypes.APPLICATION_JSON, SerializationUtil.serializeErrorWithResult(e.code, e.description, msg)))
-            else
-              Directives.complete(StatusCodes.OK, HttpEntities.create(ContentTypes.APPLICATION_JSON, SerializationUtil.serializeErrorWithResult(e.code, e.description, "")))
-          case None =>
-            Directives.complete(StatusCodes.OK, HttpEntities.create(ContentTypes.APPLICATION_JSON, SerializationUtil.serializeErrorWithResult(e.code, e.description, "")))
-        }
+      case e: ErrorResponse => {
+        val exceptionMessage: String = e.exception.map(Throwables.getStackTraceAsString).getOrElse("")
+        Directives.complete(StatusCodes.OK, HttpEntities.create(ContentTypes.APPLICATION_JSON, SerializationUtil.serializeErrorWithResult(e.code, e.description, exceptionMessage)))
+      }
+      case e: InternalErrorResponse => {
```

Comment:
can we have just `ErrorResponse` in case of `InternalErrorResponse` inheritance from `ErrorResponse`?

Reply:
done